### PR TITLE
Added filter buttons to Top31 chart

### PR DESCRIPTION
**Pull request by ericplan**
Added filter buttons to the chart
Added logic to both chart files to support this
Added bootstrap-select cdn to header.php + minor style additions 
Added language support in both ini files and the chart files 
Minor issue: button placement is not correct in Safari (few pixels up) but OK in Chrome 
HTML validated OK

### CODE REVIEW

File: ZonPHP/inc/header.php [hunk 1]
```diff
@@ -60,6 +63,18 @@
...
+	<!-- jqwidgets combobox -->
+<!-- 
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jqwidgets/18.0.0/jqwidgets/jqx-all.js" ></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jqwidgets/18.0.0/jqwidgets/jqxcore.min.js"></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jqwidgets/18.0.0/jqwidgets/jqxcolorpicker.min.js"></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jqwidgets/18.0.0/jqwidgets/jqxbuttons.min.js"></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jqwidgets/18.0.0/jqwidgets/jqxscrollbar.min.js"></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jqwidgets/18.0.0/jqwidgets/jqxlistbox.min.js"></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/jqwidgets/18.0.0/jqwidgets/jqxcombobox.min.js"></script>
+ -->
```
Comment (seehausen) [inline ZonPHP/inc/header.php]: This is commented out, if we do not need this anymore, we can remove it
(history can be found in git)

File: ZonPHP/charts/top31_chart.php [hunk 2]
```diff
@@ -11,13 +11,30 @@
...
+
 $sql = "SELECT db1.*
 FROM " . TABLE_PREFIX . "_maand AS db1 
-JOIN (SELECT Datum_Maand, sum(Geg_Maand) as mysum FROM " . TABLE_PREFIX . "_maand $whereInClause Group by Datum_Maand ORDER BY mysum $DESC_ASC LIMIT 0,31) AS db2
-ON db1.Datum_Maand = db2.Datum_Maand $whereInClause order by mysum desc";
 
+JOIN (SELECT Datum_Maand, sum(Geg_Maand) as mysum FROM " . TABLE_PREFIX . "_maand $whereInClause  AND MONTH(Datum_Maand) IN ($whereInMonth) AND YEAR(Datum_Maand) IN ($whereInYear) Group by Datum_Maand ORDER BY mysum $DESC_ASC LIMIT 0,31) AS db2
+
+ON db1.Datum_Maand = db2.Datum_Maand $whereInClause order by mysum desc";
+//echo $sql;
```
Comment (seehausen) [inline ZonPHP/charts/top31_chart.php]: Remove echo lines at least if they are commented

File: ZonPHP/inc/language/de.ini [hunk 3]
```diff
@@ -26,8 +26,10 @@ index = "Startseite"
 importer = "Importer"
 inverter = "Wechselrichter"
 jaar = "Jahr"
+jaar_sel = "Jahre selektiert"
```
Comment (seehausen) [inline ZonPHP/inc/language/de.ini]: I would call it "Jahre ausgewählt"

File: ZonPHP/inc/language/de.ini [hunk 4]
```diff
@@ -26,8 +26,10 @@ index = "Startseite"
...
 location = "Standort"
 maand = "Monat"
+maand_sel = "Monate selektiert"
```
Comment (seehausen) [inline ZonPHP/inc/language/de.ini]: I would call it "Monate ausgewählt"

File: ZonPHP/pages/top31.php [hunk 5]
```diff
@@ -10,11 +10,25 @@
...
+$ts = new DateTime($date);
+$formatter = new IntlDateFormatter($locale, IntlDateFormatter::LONG, IntlDateFormatter::LONG);
+$parts=array('M','MMM','MMMM');
+$month_local_inner= array();
+foreach($parts as $cd) {
+        $formatter->setPattern($cd);
+        $month_local_inner[] = $formatter->format($ts);
+	}
+$month_local_inner=str_replace('.','',$month_local_inner);//strips dot from abbreviation
+$month_local[]=$month_local_inner;
```
Comment (seehausen) [inline ZonPHP/pages/top31.php]: Add indent for better reading.   just reformat the code

File: ZonPHP/pages/top31.php [hunk 6]
```diff
@@ -23,19 +37,82 @@
...
+    					<select id="tea" name="roles" class="selectpicker show-tick" 
+            			title="<?= getTxt("jaar") ?>"
+            			multiple="multiple"
+            			data-width="fit"	
+	            		data-selected-text-format="count > 3"
+    	        		data-count-selected-text="{0} <?= getTxt("jaar_sel"); ?>"	 >
+      					<?php
+        				foreach($years as $item){
+            			echo "<option value='$item'>$item</option>";
+			        	}
```
Comment (seehausen) [inline ZonPHP/pages/top31.php]: indent format

**Comment by seehausen** [comment]
@ericplan check my changes, and if OK, you can merge

File: ZonPHP/pages/top31.php [hunk 7]
```diff
@@ -10,11 +10,25 @@
 
 include_once ROOT_DIR . "/inc/header.php";
 include_once "../charts/top31_chart.php";
-
 $padding = '- 0px';
 $corners = 'border-bottom-left-radius: 0px !important; border-bottom-right-radius: 0px;';
 
+$month_local = array();
+for ($nn = 1; $nn <= 12; ++$nn) {
+    $date = '2024-' . $nn . '-01'; // date only used for loop
```
Comment (seehausen) [inline ZonPHP/pages/top31.php:18]: Comment can be "year" instead of "date" but minor

**Comment by ericplan** [comment]
Hi,

This part will be replaced by a shorter and nicer PHP-function. Stil testing.

Eric

**Pull request merged**